liana-gui/src/app/state/receive.rs

```diff
@@ -453,6 +471,26 @@ impl ShowQrCodeModal {
     }
 }
 
+pub struct ShowBip21QrCodeModal {
+    qr_code: qr_code::Data,
+    bip21: String,
+}
+
+impl ShowBip21QrCodeModal {
+    pub fn new(bip21: &payjoin::Url, _index: ChildNumber) -> Option<Self> {
+        qr_code::Data::new(format!("{}", bip21))
```

Comment:
I opted to keep the bip21 qr and regular qr seperate to allow the Url to be passed and to keep the regular qr unchanged accepting an address, do you think we should unify it?